### PR TITLE
Proposal: secrets,configmap,downwardAPI all in one volume

### PR DESCRIPTION
**Pull request by jpeeler**
Copied from kubernetes/kubernetes#35313 (with slight modifications)

### CODE REVIEW

**Comment by philips** [comment]
I agree that this should be done. And looking back it looks like the discussion largely agreed as well but we are stuck on how to implement.

**Comment by pmorie** [comment]
@philips I don't think we are stuck on how to implement, really.  The existing plugins all share the majority of their guts via `AtomicWriter`

@jpeeler @thockin do we have the implementation-sharing bits ironed out or is that just me?

**Comment by thockin** [comment]
I don't think they are ironed out, but they seem straight-forward enough to
PoC.

**Comment by pmorie** [comment]
@thockin what gaps do you think we need to close to get this proposal merged?

File: contributors/design-proposals/all-in-one-volume.md [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+    DownwardAPI *DownwardAPIVolumeSource `json:"downwardAPI,omitempty"`
+}
+```
+
+### Additional required modifications
+
+Add to the VolumeSource struct:
+
+```go
+SystemProjection *SystemProjections `json:"system,omitempty"`
```
Comment (thockin) [inline contributors/design-proposals/all-in-one-volume.md]: I dislike "system" as a name - how about "projected" or "virtual" or "from" or something that is more explanatory

**Comment by thockin** [comment]
It's assigned to you, @pmorie so you have final say.

**Comment by pmorie** [comment]
@thockin, I don't think 'system' is a great name and would be cool bikeshedding on the name a little before we have @jpeeler get that much further.  I think he's in a place where it should be easy to finish up by code cut.  Any thoughts about a better name?

**Comment by thockin** [comment]
"system" is a terrible name.  I offered "projected" or "virtual" ?

**Comment by pmorie** [comment]
@thockin I like 'projected', 'virtual' feels a little too general to me.

@jpeeler, any objection to `projected` for the driver name?

**Comment by jpeeler** [comment]
@pmorie "projected" sounds fine to me!

**Comment by pmorie** [comment]
@jpeeler okay - update the proposal and i'll merge.

**Pull request merged**